source/plugin/advanced/debugging.rst

```diff
@@ -0,0 +1,119 @@
+================================
+Plugin Debugging and Hotswapping
+================================
+
+Every developer, at some point, has had the *pleasure* of debugging their plugin.
+Generally speaking, debugging is done by throwing a multitude of logs in specific areas to narrow down breakpoints,
+compiling the plugin, moving the result to the test folder, and running the server. Along that process, praying to
+whatever deity you have that everything doesn't immediately fail and make the past hour or so of programming in vain.
```

Comment:
Quibble. We try not to make references to religion, it can generate friction (even as a light-hearted joke). Just sayin'. I can live with the rest, hyperbolic as it is ;)